### DISCUSSION TITLE
Restore-DbaDbSnapshot error handling

### DISCUSSION THREAD

**Discussion by strange-v** [dataplat/dbatools#7399] [Q&A] [answered]
Hello.
I use the latest version of dbatools (1.0.153) to restore a database snapshot and it is unclear to me how to correctly handle errors.

For example, I use this command:
`Restore-DbaDbSnapshot -SqlInstance localhost -Database DBName -Snapshot IncorrectSnapshotName -Force`

it executes without any error or return result. How to handle the case when the database wasn't restored? I tried to add -EnableException but nothing changed.

It seems, that I'm missing something quite obvious, but it isn't clear what exactly.

PS
Of course, the main goal is to handle a different error (e.g. the database is in an incorrect state and restore will fail), an incorrect snapshot name is just a way to generate an error.

**Comment by andreasjordan** [comment, marked as answer]
Have a look at the source code. The command first runs this: `Get-DbaDbSnapshot -SqlInstance $server -Database $Database -ExcludeDatabase $ExcludeDatabase -Snapshot $Snapshot`.
With an incorrect snapshot name, the Get- command does not return anything. That is by design. If you want to make sure that there is a snapshot with that name, you have to use `Get-DbaDbSnapshot` yourself and test if there is a result.

**Reply by strange-v** [reply]
Thanks, haven't noticed that.
